tests/test_database.py

```diff
@@ -0,0 +1,94 @@
+from os import remove
+from os.path import isfile
+
+import unittest
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from space_allocator.app import (dojo, database, Staff, Fellow,
+	LivingSpace, Office, Base, RoomDB, PersonDB)
+
+class TestSaveState(unittest.TestCase):
+
+	def setUp(self):
+		self.dojo = dojo
+		self.database = database
+		self.offices = ["Dede", "Didi", "Dodo", "Dudu", "Fafa", "Fefe"]
+		self.livingspaces = ["Bebe", "Bibi", "Bobo", "Bubu", "Caca", "Cefe"]
+	
+	def test_save_state_creates_file_when_db_is_defined(self):
+		self.database.save_state("trial.db")
+		self.assertTrue(isfile("trial.db"))
+		remove("trial.db")
+
+	def test_save_state_saves_data_succesfully_with_defined_db(self):
+		#First, create two rooms with guaranteed allocations
+		self.dojo.create_room("Dada", "Office")
+		self.dojo.create_room("Baba", "livingspace")
+		self.dojo.load_people("dummy.txt")
+		#create more rooms of which Fefe and Cece will be unallocated so...
+		#...that we can test whether the function works for empty rooms
+		for office in self.offices:
+			self.dojo.create_room(office, "Office")
+		for livingspace in self.livingspaces:
+			self.dojo.create_room(livingspace, "livingspace")
+		self.database.save_state("trial.db")
+		engine = create_engine("sqlite:///trial.db", echo = False)
+		Base.metadata.bind = engine
+		DBSession = sessionmaker(bind=engine)
+		session = DBSession()
+		dbrooms = session.query(RoomDB).all()
+		self.assertEqual(len(dbrooms), 14)
+		dbpersons = session.query(PersonDB).all()
+		self.assertEqual(len(dbpersons), 35)
+		self.assertEqual(dbpersons[17].name, "Skip Cheru")
+
+	def test_save_state_saves_data_succesfully_with_undefined_db(self):
+		self.dojo.load_people("dummy.txt")
+		self.database.save_state()
+		engine = create_engine("sqlite:///dojo.db", echo = False)
+		Base.metadata.bind = engine
+		DBSession = sessionmaker(bind=engine)
+		session = DBSession()
+		dbrooms = session.query(RoomDB).all()
```

Comment:
For L49-52 Same as the comment above to avoid repetition. That's why there is the setUp method.